homeassistant/components/egardia.py

```diff
@@ -0,0 +1,120 @@
+D_EGARDIASYS = 'egardiadevice'
+D_EGARDIANM = 'egardianame'
+D_EGARDIARSENABLED = 'egardia_rs_enabled'
+D_EGARDIARSCODES = 'egardia_rs_codes'
+D_EGARDIADEV = 'egardia_dev'
+NOTIFICATION_ID = 'egardia_notification'
+NOTIFICATION_TITLE = 'Egardia'
+ATTR_DISCOVER_DEVICES = 'egardia_sensor'
+STATES = {
+    'ARM': STATE_ALARM_ARMED_AWAY,
+    'DAY HOME': STATE_ALARM_ARMED_HOME,
+    'DISARM': STATE_ALARM_DISARMED,
+    'HOME': STATE_ALARM_ARMED_HOME,
+    'TRIGGERED': STATE_ALARM_TRIGGERED,
+    'UNKNOWN': STATE_UNKNOWN,
+}
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_VERSION, default=DEFAULT_VERSION): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_REPORT_SERVER_CODES): vol.All(cv.ensure_list),
+        vol.Optional(CONF_REPORT_SERVER_ENABLED,
+                     default=DEFAULT_REPORT_SERVER_ENABLED): cv.boolean,
+        vol.Optional(CONF_REPORT_SERVER_PORT,
+                     default=DEFAULT_REPORT_SERVER_PORT): cv.port,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Egardia platform."""
+    from pythonegardia import egardiadevice
+    from pythonegardia import egardiaserver
+    domainconf = config[DOMAIN]
+    username = domainconf.get(CONF_USERNAME)
+    password = domainconf.get(CONF_PASSWORD)
+    host = domainconf.get(CONF_HOST)
+    port = domainconf.get(CONF_PORT)
+    version = domainconf.get(CONF_VERSION)
+    rs_enabled = domainconf.get(CONF_REPORT_SERVER_ENABLED)
+    rs_port = domainconf.get(CONF_REPORT_SERVER_PORT)
+    try:
+        device = hass.data[D_EGARDIASYS] = egardiadevice.EgardiaDevice(
+            host, port, username, password, '', version)
+    except requests.exceptions.RequestException:
+        return False
+    except egardiadevice.UnauthorizedError:
+        _LOGGER.error("Unable to authorize. Wrong password or username")
+        return False
+    # Set up the egardia server if enabled
+    if rs_enabled:
+        _LOGGER.info("Setting up EgardiaServer")
+        try:
+            if D_EGARDIASRV not in hass.data:
+                server = egardiaserver.EgardiaServer('', rs_port)
+                bound = server.bind()
+                if not bound:
+                    raise IOError("Binding error occurred while " +
+                                  "starting EgardiaServer")
+                hass.data[D_EGARDIASRV] = server
+                server.start()
+        except IOError:
+            return
+
+    discovery.load_platform(hass, 'alarm_control_panel', DOMAIN,
+                            discovered=domainconf, hass_config=config)
+
+    def handle_stop_event(event):
+        """Callback function for HA stop event."""
+        hass.data[D_EGARDIASRV].stop()
```

Comment:
Move this function under the `rs_enabled` if block above and call `server.stop`.

Reply:
I do not get this. This `handle_stop_event` is supposed to be the callback for the `EVENT_HOMEASSISTANT_STOP` event (it is wired to that event just two lines down). If I move it as you suggested, it is not a callback anymore, right?

Reply:
Done.